Dockerfile

```diff
@@ -0,0 +1,83 @@
+# Use an x86_64 specific base image
+FROM --platform=linux/amd64 ubuntu:20.04
+
+# Set up environment
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PYTHONUNBUFFERED=1
+WORKDIR /workspace
+
+# Install system dependencies
+RUN apt-get update && apt-get install -y \
+    software-properties-common && \
+    add-apt-repository ppa:deadsnakes/ppa && \
+    apt-get update && apt-get install -y \
```

Comment:
Let's consider removing the duplicate `apt-get update` here, perhaps as part of an effort to split the RUN command into two smaller ones (see above).